lib/test/mock_loader.js

```diff
@@ -23,6 +20,10 @@ let debug = require('internal/util/debuglog').debuglog('test_runner', (fn) => {
   debug = fn;
 });
 const { createRequire, isBuiltin } = require('module');
+const { defaultResolve } = require('internal/modules/esm/resolve');
+const {
+  defaultGetFormatWithoutErrors,
+} = require('internal/modules/esm/get_format');
```

Comment:
It feels wrong to me that the mock loader would be using module internals directly rather than the public API. This interdependence will make it harder to refactor modules stuff since non-public changes might break the test runner.

Reply:
As far as I understand, mock_loader is an internal module resolver when using mock module. So this has to decide the format internally, is there an public API that can solve it? I'm sorry if it's something I haven't found. Thanks!

Reply:
You should be able to use the next hook in the chain, I'd think.